RecoHGCal/TICL/plugins/TrackstersProducer.cc

```diff
@@ -43,8 +50,29 @@ class TrackstersProducer : public edm::stream::EDProducer<> {
 };
 DEFINE_FWK_MODULE(TrackstersProducer);
 
-TrackstersProducer::TrackstersProducer(const edm::ParameterSet& ps)
-    : myAlgo_(std::make_unique<PatternRecognitionbyCA>(ps)) {
+std::unique_ptr<TrackstersCache> TrackstersProducer::initializeGlobalCache(const edm::ParameterSet& params) {
+  // this method is supposed to create, initialize and return a TrackstersCache instance
+  TrackstersCache* cache = new TrackstersCache(params);
```

Comment:
don't use `new`. Just define the `unique_ptr` right away:
```
auto cache = std::make_unique<TrackstersCache>(params);
```

Reply:
Ok, will add that in the next commit.